webapp/src/shared/components/Timer/Timer.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react';
+import { Text, TextVariant } from '../index';
+import './Timer.css';
+
+type TimerProps = {
+  shouldRun?: boolean;
+  textVariant?: TextVariant;
+  timeString?: string;
+  isBackwardsCount?: boolean;
+  onTimeOut?: () => void;
+};
+
+type clockType = {
+  timeInSeconds: number;
+  hours: number;
+  minutes: number;
+  seconds: number;
+};
+
+const secondsToClockType = (seconds: number): clockType => {
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor(seconds / 60) - hours * 60;
+  const lseconds = Math.floor(seconds % 60);
+  return {
+    timeInSeconds: seconds,
+    hours,
+    minutes,
+    seconds: lseconds,
+  };
+};
+
+const parseTime = (input: string): clockType => {
+  let a = input.split(':'); // split it at the colons
+  if (a[0] && a[1] && a[2]) {
+    return {
+      timeInSeconds:
+        parseInt(a[0]) * 60 * 60 + parseInt(a[1]) * 60 + parseInt(a[2]),
+      hours: parseInt(a[0]),
+      minutes: parseInt(a[1]),
+      seconds: parseInt(a[2]),
+    };
+  }
+  throw new Error('Bad time format');
+};
+
+export default function Timer({
+  shouldRun = true,
+  textVariant = TextVariant.TITLE,
+  timeString = '00:01:00',
```

Comment:
What do you think about better exposing a `Number` property , for example `timeInSeconds`? This way we could easily reuse this component as mentioned in other comment and avoid parsing

Reply:
But why would we want to avoid parsing? The point is that a user can write down a time interval of its liking, and have a timer set up in place for the appropriate amount of time. I don't see a way around parsing. Maybe we can push the parsing up into some of the components/pages above this one, and only render this after the input has been validated and converted to seconds as a number, so that we don't have to manage a possible throw inside the render method of a JSX.Element. Since we don't have a page that uses this yet, I'll still leave it like this for the time being.